Implementation of the function to add BBMASK

terasawa-lin opened the pull request:
# Purpose of pull request

setup-emlinux reads add_BBMASK.txt and writes it to conf/local.conf.

The following packages are listed in add_BBMASK.txt.
 poky/meta/recipes-bsp/grub/grub_2.02.bb
 poky/meta/recipes-bsp/grub/grub-efi_2.02.bb
 poky/meta/recipes-sato/webkit/webkitgtk_2.22.7.bb
 poky/meta/recipes-graphics/xorg-lib/libxxf86misc_1.0.4.bb
 meta-debian/recipes-debian/fcode-utils/fcode-utils_debian.bb

hiraku-wfs commented on scripts/setup-emlinux:
```diff
@@ -55,6 +55,7 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     # use poky's base-files package temporary.
     echo "BBMASK .= \"|meta-debian/recipes-debian/base-files/base-files_debian.bb\"" >> conf/local.conf
     echo "DL_DIR = \"\${TOPDIR}/../downloads\"" >> conf/local.conf
+    cat ../repos/meta-emlinux/scripts/add_BBMASK.txt >> conf/local.conf
```
Please use ${REPOS} instead of "../repos".

hiraku-wfs commented on scripts/add_BBMASK.txt:
```diff
@@ -0,0 +1,5 @@
+           poky/meta/recipes-bsp/grub/grub-efi_2.02.bb \
+           poky/meta/recipes-sato/webkit/webkitgtk_2.22.7.bb \
+           poky/meta/recipes-graphics/xorg-lib/libxxf86misc_1.0.4.bb \
+           meta-debian/recipes-debian/fcode-utils/fcode-utils_debian.bb"
```
This file is not a program but a configuration file.
Please rename add_BBMASK.txt to bbmask.conf.

hiraku-wfs commented on scripts/setup-emlinux:
```diff
@@ -55,6 +55,7 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     # use poky's base-files package temporary.
     echo "BBMASK .= \"|meta-debian/recipes-debian/base-files/base-files_debian.bb\"" >> conf/local.conf
```
Please include these mask configs in bbmask.conf.

hiraku-wfs reviewed (commented):
This single commit does ...
- Separate BBMASK config from setup-emlinux
- Mask recipes which are unbuildable and unused in EMLinux

Please split it into two commits and add appropriate subjects to each commit.

terasawa-lin commented:
Fixed.
Please let us know if you have any mistakes.

terasawa-lin commented:
There was a mistake.

terasawa-lin commented:
Fixed the mistake

hiraku-wfs commented on scripts/setup-emlinux:
```diff
@@ -50,11 +50,10 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     echo "PACKAGE_CLASSES = \"package_deb\"" >> conf/local.conf
-    echo "BBMASK = \"meta-debian/recipes-core\"" >> conf/local.conf
     # nsswitch.conf is not provided by debian base-file package but systemd requires it.
     # use poky's base-files package temporary.
```
Please move this comment to the bbmask.conf.

hiraku-wfs commented on scripts/setup-emlinux:
```diff
@@ -50,11 +50,10 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     # use poky's base-files package temporary.
-    echo "BBMASK .= \"|meta-debian/recipes-debian/base-files/base-files_debian.bb\"" >> conf/local.conf
     echo "DL_DIR = \"\${TOPDIR}/../downloads\"" >> conf/local.conf
+    cat ${REPOS}/meta-emlinux/scripts/bbmask.conf >> conf/local.conf
```
This commit should create the bbmask.conf, which includes mask-configs about the recipes-core and the base-files.
In your next commit, please add the rest configs.

terasawa-lin replied:
Is this comment as a commit message for setup-emlinux?

terasawa-lin committed:
Separate BBMASK config from setup-emlinux

Items of bbmask-ed recipes may change. So we separate the BBMASK list into
the new file bbmask.conf.

terasawa-lin committed:
Add unbuildable and unused recipes to bbmask config

terasawa-lin commented:
Fixed.

hiraku-wfs commented:
Thanks!
